ad-joining/register-computer/ad/domain.py

```diff
@@ -212,6 +213,53 @@ def get_upn_by_samaccountname(self, samaccountname):
         else:
             return self.__to_scalar(self.__connection.entries[0]["userPrincipalName"])
 
+    def get_group(self, group_name, search_base_dn):
+        try:
+            self.__connection.search(
+                search_filter="(&(objectClass=group)(cn=%s))" % group_name,
+                search_base=search_base_dn,
+                attributes=[
+                    "distinguishedName",
+                    "name",
+                    ActiveDirectoryConnection.LDAP_ATTRIBUTE_PROJECT_ID])
+
+            return [Group(
+                    entry.entry_dn,
+                    self.__to_scalar(entry["name"]),                    
+                    self.__to_scalar(entry[ActiveDirectoryConnection.LDAP_ATTRIBUTE_PROJECT_ID]))
+                for entry in self.__connection.entries]
+        except ldap3.core.exceptions.LDAPNoSuchObjectResult as e:
+            raise NoSuchObjectException(e)
+
+    def add_group(self, ou, group_name):
+        try:           
+            dn = "CN=%s,%s" % (group_name, ou)
+            self.__connection.add(
+                dn,
+                [
+                "group",
```

Comment:
You're not specifying the `groupType` - does this still create a domain-local security group?

Reply:
It was created as a global security group. I added groupType and set it domain-local security group.